.github/workflows/gh-pages.yaml

```diff
@@ -35,4 +35,61 @@ jobs:
         if: ${{ github.ref == 'refs/heads/master' }}
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./book/book
\ No newline at end of file
+          publish_dir: ./book/book
+
+  benchmark:
+    needs: deploy
+    # continuous-benchmark requires sensible permissions, hence why it should never be run on PRs
+    if: github.event_name != 'pull_request'
+    permissions:
+      # deployments permission to deploy GitHub pages website
+      deployments: write
+      # contents permission to update benchmark contents in gh-pages branch
+      contents: write
+    runs-on: ubuntu-20.04
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: 1.64.0
+          override: true
+          components: rustfmt, clippy
+      
+      # Install benchmark jobs deps
+      - uses: dart-lang/setup-dart@v1
+        with:
+          sdk: '2.17.6' # TODO upgrade later #655
+      
+      - name: Install dart dependencies (benchmarks)
+        working-directory: ./frb_example/benches/dart
+        run: dart pub get
+      
+      # Prepare benchmark first
+      - name: Prepare benchmark
+        run: cd frb_example/benches/rust && cargo build --release
+      
+      # Run benchmark and stores the output to a file
+      - name: Run benchmark
+        run: |
+          mkdir -p ./book/benches && \
+          touch ./book/benches/output.txt && \
+          dart frb_example/benches/dart/lib/main.dart target/release/libflutter_rust_bridge_example_benchmark_suite.so | tee ./book/benches/output.txt
+        env:
+          JSON: true
+
+      # Update benchmark and deploy to GitHub Pages
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: Flutter Rust Bridge Benchmark
+          tool: 'customSmallerIsBetter'
+          output-file-path: ./book/benches/output.txt
```

Comment:
it's to avoid deploying multiple times, but they can be split if we prevent the 2 workflow from running concurrently (otherwise I guess from time to time there's gonna be file conflicts).

Reply:
Indeed I still suspect it will be deployed multiple times. Since they are run parallel.

Reply:
> Indeed I still suspect it will be deployed multiple times.

not if `auto-push` is set to false I guess (see [here](https://github.com/benchmark-action/github-action-benchmark#charts-on-github-pages-1)). then there's [concurrency](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idconcurrency) in github worfklow syntax.

Reply:
> not if auto-push is set to false I guess (see [here](https://github.com/benchmark-action/github-action-benchmark#charts-on-github-pages-1)). then there's [concurrency](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idconcurrency) in github worfklow syntax.

worth giving a try ?

Reply:
Not sure. The problematic case I have in my mind is:

1. push some code, gh-pages is triggered
2. the benchmark is run, and new code is commited to main branch (i.e. that output txt)
3. that new code triggers a new gh-pages workflow
4. ... even dead loop

Reply:
In this case running both actions in the same workflow (with `continuous-benchmark`'s `auto-push` set to `false`) would guarantee that both outcomes get deployed together by `gh-pages`

Reply:
Then seems we need to merge them into one *job* instead of two jobs?

Anyway IMHO we may need to let web bench and non-web bench be consistent. Maybe both in ci.yaml for simplicity?